pkg/cmd/roachtest/test_runner.go

```diff
@@ -2116,3 +2134,47 @@ func getTestParameters(t *testImpl, c *clusterImpl, createOpts *vm.CreateOpts) m
 
 	return clusterParams
 }
+
+// PollPreemptionInterval is how often to poll for preempted VMs.
+var PollPreemptionInterval = 5 * time.Minute
+
+var preemptionMonitor = monitorForPreemptedVMs
+
+func monitorForPreemptedVMs(
+	ctx context.Context,
+	t test.Test,
+	c cluster.Cluster,
+	l *logger.Logger,
+	preemptedVMCh chan vm.PreemptedVM,
+) {
+	if c.IsLocal() || !c.Spec().UseSpotVMs {
+		close(preemptedVMCh)
+		return
+	}
+
+	go func() {
+		pollCtx, cancel := context.WithCancel(ctx)
+		defer cancel()
+		defer close(preemptedVMCh)
+
+		for {
+			select {
+			case <-pollCtx.Done():
+				return
+			case <-time.After(PollPreemptionInterval):
+				preemptedVMs, err := c.GetPreemptedVMs(ctx, l)
+				if err != nil {
+					l.Printf("monitorForPreemptedVMs: failed to check preempted VMs:\n%+v", err)
+					continue
+				}
+
+				if len(preemptedVMs) != 0 {
+					for _, m := range preemptedVMs {
+						preemptedVMCh <- m
+					}
+					t.Errorf("monitorForPreemptedVMs: Preempted VMs detected: %s", preemptedVMs)
```

Comment:
Just as a safeguard, we could swap the order with above. I see that the buffered channel is properly sized to allow for at most messages as there are nodes. However, what if for some reason, `GetPreemptedVMs` returns more? In that case, the above will block, effectively preventing to bail out early.

Reply:
Removed the channel logic so I believe this no longer applies.